core/server/src/test/java/alluxio/master/journal/JournalWriterTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import alluxio.master.journal.JournalWriter.EntryOutputStream;
+import alluxio.proto.journal.Journal.JournalEntry;
+import alluxio.underfs.UnderFileSystem;
+
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Unit tests for {@link JournalWriter}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(JournalWriter.class)
+public class JournalWriterTest {
+
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  private Journal mJournal;
+
+  @Before
+  public void before() throws Exception {
+    File journalFolder = mFolder.newFolder();
+    mJournal = new ReadWriteJournal(journalFolder.getAbsolutePath());
+  }
+
+  @Test
+  public void rotateLogOnFlushIOException() throws Exception {
```

Comment:
There are a few differences that make it more complex to factor out common pieces. I think the methods are currently short and simple enough that refactoring would end up making them less readable.